dotnet/src/Connectors/Connectors.AI.Oobabooga/TextCompletion/OobaboogaTextCompletion.cs

```diff
@@ -213,25 +212,27 @@ public sealed class OobaboogaTextCompletion : ITextCompletion
             httpRequestMessage.Headers.Add("User-Agent", HttpUserAgent);
 
             using var response = await this._httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
-            response.EnsureSuccessStatusCode();
 
-            var body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
 
-            TextCompletionResponse? completionResponse = JsonSerializer.Deserialize<TextCompletionResponse>(body);
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException e)
+            {
+                throw new HttpOperationException(response.StatusCode, responseContent, e.Message, e);
+            }
```

Comment:
> Can we implement all of this in an extension method, so we're not always try/catch/throwing? Something like: public static void EnsureSuccess(this HttpResponseMessage, string? customContent, ILogger? logger)

Good idea, done.

Reply:
> Also, I think you'll get a performance boost by checking for if (response.IsSuccess...) instead of try/catching. @stephentoub is that true?

Quick search in .net runtime repo shows that the check is already there:
<img width="227" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/d63d2b31-26d4-476e-a741-c22f257756a1">

Not sure if it's relevant to all versions of .net runtime, though.

Reply:
The purpose of EnsureSuccessStatusCode is to throw an exception if it's not successful, and it does that by checking the status code, in all versions. If you want to avoid an extra throw/catch, you can check the status code yourself instead of using EnsureSuccessStatusCode and just directly throw the HttpOperationException.

Reply:
There's no goal to avoid the throw/catch approach unless it's impacting performance, which is not the case here. The EnsureSuccessStatusCode method produces the HttpRequestException in case of an error, and we save it as an internal exception of the HttpOperationException. This approach might be useful to get additional details if needed in case they are not available through HttpOperationException properties for some reason.